integration-tests/contracts/ethereum_vrf_common.go

```diff
@@ -106,6 +107,29 @@ func parseRequestRandomnessLogs(coordinator Coordinator, logs []*types.Log) (*Co
 	return randomWordsRequestedEvent, nil
 }
 
+func ParseRandomWordsFulfilledLogs(coordinator Coordinator, logs []*types.Log) ([]*CoordinatorRandomWordsFulfilled, error) {
+	var randomWordsFulfilledEventArr []*CoordinatorRandomWordsFulfilled
+	for _, eventLog := range logs {
+		for _, topic := range eventLog.Topics {
+			if topic.Cmp(vrf_coordinator_v2_5.VRFCoordinatorV25RandomWordsFulfilled{}.Topic()) == 0 {
+				randomWordsFulfilledEvent, err := coordinator.ParseRandomWordsFulfilled(*eventLog)
+				if err != nil {
+					return nil, fmt.Errorf("parse RandomWordsFulfilled log failed, err: %w", err)
+				}
+				randomWordsFulfilledEventArr = append(randomWordsFulfilledEventArr, randomWordsFulfilledEvent)
+			}
+			if topic.Cmp(vrf_coordinator_v2.VRFCoordinatorV2RandomWordsFulfilled{}.Topic()) == 0 {
+				randomWordsFulfilledEvent, err := coordinator.ParseRandomWordsFulfilled(*eventLog)
+				if err != nil {
+					return nil, fmt.Errorf("parse RandomWordsFulfilled log failed, err: %w", err)
+				}
+				randomWordsFulfilledEventArr = append(randomWordsFulfilledEventArr, randomWordsFulfilledEvent)
+			}
```

Comment:
That looks like the same code to me. Perpahs just checking if the topic comes from V2 or V25 and if that's not the case, throw an error? So something like this:

```
if topic.Cmp(vrf_coordinator_v2_5.VRFCoordinatorV25RandomWordsFulfilled{}.Topic()) != 0 && topic.Cmp(vrf_coordinator_v2.VRFCoordinatorV2RandomWordsFulfilled{}.Topic()) != 0 {
	// return an error
}
// topic is detected, the processing is the same for both coordinator versions
randomWordsFulfilledEvent, err := coordinator.ParseRandomWordsFulfilled(*eventLog)
if err != nil {
	return nil, fmt.Errorf("parse RandomWordsFulfilled log failed, err: %w", err)
}
randomWordsFulfilledEventArr = append(randomWordsFulfilledEventArr, randomWordsFulfilledEvent)
```

Reply:
good catch, thanks!